src/app/functions/server/updateDataBrokerScanProfile.test.ts

```diff
@@ -0,0 +1,166 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { it, expect, jest } from "@jest/globals";
+import { OnerepProfileRow, UpdateableProfileDetails } from "knex/types/tables";
+
+const currentProfileDetails: OnerepProfileRow = {
+  id: 1234,
+  onerep_profile_id: 5678,
+  first_name: "First01",
+  last_name: "last01",
+  middle_name: "",
+  first_names: [],
+  middle_names: [],
+  last_names: [],
+  city_name: "City01",
+  state_code: "NY",
+  addresses: [{ city: "City01", state: "NY" }],
+  phone_numbers: ["8005553534"],
+  date_of_birth: new Date(),
+  created_at: new Date(),
+  updated_at: new Date(),
+};
+
+const newProfileDetails: UpdateableProfileDetails = {
+  first_name: "First01",
+  last_name: "last01",
+  middle_name: "",
+  first_names: [],
+  middle_names: [],
+  last_names: [],
+  addresses: [{ city: "City01", state: "NY" }],
+  phone_numbers: ["8005553534"],
+};
+
+jest.mock("./onerep", () => ({
+  updateProfile: jest.fn(),
+}));
+
+jest.mock("./refreshStoredScanResults");
+
+jest.mock("./logging", () => {
+  return {
+    logger: {
+      info: jest.fn(),
+    },
+  };
+});
+
+beforeEach(() => {
+  jest.spyOn(console, "info").mockImplementation(() => undefined);
+});
+
+describe("Update broker scan profile", () => {
+  it("updates the profile details", async () => {
+    jest.mock("../../../db/tables/onerep_profiles", () => ({
+      getProfileDetails: jest.fn(() => Promise.resolve(currentProfileDetails)),
+      updateProfileDetails: jest.fn(),
+    }));
+    const updateDataBrokerScanProfile = (
+      await import("./updateDataBrokerScanProfile")
+    ).default;
+
+    expect(async () => {
+      await updateDataBrokerScanProfile(5678, {
+        ...newProfileDetails,
+        first_names: ["First02", "First03"],
+        last_names: ["Last02"],
+        middle_name: "Middle01",
+        middle_names: ["Middle02", "Middle03"],
+      });
+    }).not.toThrow();
+  });
+
+  it("throws an error when the profile has no profile ID assigned", async () => {
+    jest.mock("../../../db/tables/onerep_profiles", () => ({
+      getProfileDetails: jest.fn(() =>
+        Promise.resolve({
+          ...currentProfileDetails,
+          onerep_profile_id: null,
+        }),
+      ),
+      updateProfileDetails: jest.fn(),
+    }));
+    const updateDataBrokerScanProfile = (
+      await import("./updateDataBrokerScanProfile")
+    ).default;
+
+    await expect(async () => {
+      await updateDataBrokerScanProfile(5678, {
+        ...newProfileDetails,
+        first_names: ["First02", "First03"],
+        last_names: ["Last02"],
+        middle_name: "Middle01",
+        middle_names: ["Middle02", "Middle03"],
+      });
```

Comment:
Suggestion: these don't seem relevant for what we're testing for here?

```suggestion
      await updateDataBrokerScanProfile(5678, newProfileDetails);
```

Reply:
No, not strictly relevant: [29a8735](https://github.com/mozilla/blurts-server/pull/5660/commits/29a87351f040ed56ce8aef07cc406d76d80089cf).